modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceProductOptions.testcase

```diff
@@ -1013,6 +1013,156 @@ definition {
+		}
+	}
+
+	@description = "COMMERCE-11494. Can delete product option with all skus generated"
+	@priority = 4
+	test CanDeleteProductOptionWithGeneratedSkus {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+		}
+
+		task ("And a product option is created") {
+			CommerceProducts.openProductsAdmin();
+
+			CommerceProducts.searchAndGoToProductEntry(entryProductName = "ABS Sensor");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			CommerceProductOptions.createProductOption(key_optionName = "Option1");
+
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			CommerceProductOptions.editProductOption(
+				key_fieldType = "Select from List",
+				key_option = "Option1",
+				productOptionFieldType = "true",
+				productOptionRequired = "true",
+				productOptionSkuContributor = "true");
+		}
+
+		task ("And 2 values are added to the option") {
+			for (var valueName : list "Value1,Value2") {
+				CommerceProductOptions.createProductOptionValue(key_name = ${valueName});
+
+				SelectFrame.selectFrameTop();
+
+				SelectFrame.selectFrameNoLoading(locator1 = "CommerceEntry#IFRAME_SIDE_PANEL");
+			}
+
+			SelectFrameTop();
+
+			Click(locator1 = "Icon#CLOSE");
+		}
+
+		task ("And from the SKUs tab, plus button is clicked, Generate all SKUs combination button is selected") {
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			Click(locator1 = "Button#PLUS");
+
+			Click(
+				key_fieldName = "Generate All SKU Combinations",
+				locator1 = "FormViewBuilder#OBJECT_FIELD_BUTTON_DROPDOWN");
+		}
+
+		task ("When the option is deleted") {
+			CommerceEntry.gotoMenuTab(menuTab = "Options");
+
+			SelectFrameTop();
+
+			Click(
+				key_value = "Option1",
+				locator1 = "CommerceEntry#BODY_VERTICAL_ELLIPSIS");
+
+			CommerceEntry.clickNoError(menuItem = "Delete");
+		}
+
+		task ("Then when the admin goes back to the products admin page and the product is displayed properly") {
+			CommerceProducts.openProductsAdmin();
+
+			Pagination.viewResults(results = "Showing 1 to 10 of 50 entries.");
+
+			AssertElementPresent(locator1 = "//a[contains(text(),'ABS Sensor')]");
+
+			AssertElementPresent(locator1 = "//a[contains(text(),'Calipers')]");
+
+			AssertElementPresent(locator1 = "//a[contains(text(),'Hoses')]");
+
+			CommerceEntry.gotoMenuTab(menuTab = "SKUs");
+
+			CommerceProducts.searchProduct(search = "Value1");
+
+			AssertElementPresent(
+				key_entryTitle = "VALUE1",
+				key_label = "Inactive",
+				locator1 = "CommerceEntry#SKU_STATUS_PRODUCT_ADMIN_PAGE");
+
+			CommerceProducts.searchProduct(search = "Value2");
+
+			AssertElementPresent(
+				key_entryTitle = "VALUE2",
+				key_label = "Inactive",
+				locator1 = "CommerceEntry#SKU_STATUS_PRODUCT_ADMIN_PAGE");
```

Comment:
Please use a FOR cycle to do the checks.
The `key_entryTitle` value inside the assert and the value inside the search are the same. The only difference is that inside the macro is in uppercase.
For the cycle, you can use the `StringUtil.upperCase` to transform the variable in all uppercase characters

Take a look inside the code to understand how it is working.